packages/tuono-router/src/components/NotFound.spec.tsx

```diff
@@ -1,114 +1,86 @@
-    <a {...props} />
-  ),
+vi.mock('../components/RouterContext', () => ({
+  useRouterContext: vi.fn(),
 }))
-
-vi.mock('../hooks/useServerPayloadData.ts', () => ({
-  useServerPayloadData: (): { data: unknown; isLoading: boolean } => {
-    return {
-      data: undefined,
-      isLoading: false,
-    }
-  },
+vi.mock('./RouteMatch', () => ({
+  RouteMatch: vi.fn(),
 }))
-
-const { useRouterContext } = vi.hoisted(() => {
-  return { useRouterContext: vi.fn() }
-})
-
-vi.mock('../components/RouterContext', () => ({
-  useRouterContext,
+vi.mock('./NotFoundDefaultContent', () => ({
+  NotFoundDefaultContent: vi.fn(),
 }))
 
+interface RouterMock {
+  router: Pick<RouterInstanceType, 'routesById'>
+}
+const useRouterContextMock = vi.mocked(useRouterContext as () => RouterMock)
+const RouteMatchMock = vi.mocked(RouteMatch)
+const NotFoundDefaultContentMock = vi.mocked(NotFoundDefaultContent)
+
 describe('<NotFound />', () => {
-  afterEach(cleanup)
+  afterEach(() => {
+    cleanup()
+    vi.resetAllMocks()
+  })
+
+  const root = new Route({
+    isRoot: true,
+    component: (({ children }: { children: ReactNode }) => (
+      <div>{children}</div>
+    )) as unknown as RouteComponent,
+  })
 
   describe('when a custom 404 page exists', () => {
     it('should render the custom 404 page', () => {
-      useRouterContext.mockReturnValue({
+      const customRoute404 = new Route({
+        getParentRoute: (): Route => root,
+        component: vi.fn() as unknown as RouteComponent,
+      })
+
+      useRouterContextMock.mockReturnValue({
         router: {
           routesById: {
-            '/404': new Route({
-              getParentRoute: (): Route => root,
-              component: createRouteComponent('404', () => (
-                <div data-testid="404">custom 404</div>
-              )),
-            }),
+            '/404': customRoute404,
             __root__: root,
           },
         },
       })
+
       render(<NotFound />)
-      expect(screen.getByTestId('root')).toMatchInlineSnapshot(
-        `
-        <div
-          data-testid="root"
-        >
-          <div
-            data-testid="404"
-          >
-            custom 404
-          </div>
-        </div>
-        `,
+
+      expect(RouteMatchMock).toHaveBeenCalledExactlyOnceWith(
+        { route: customRoute404, serverInitialData: {} },
+        undefined, // deprecated react context parameter
       )
+      expect(NotFoundDefaultContentMock).not.toHaveBeenCalled()
     })
   })
 
   describe('when a custom 404 page does not exist', () => {
     it('should render the default 404 page, wrapped by the root __layout', () => {
-      useRouterContext.mockReturnValue({
+      useRouterContextMock.mockReturnValue({
         router: {
           routesById: {
             __root__: root,
           },
         },
       })
+
       render(<NotFound />)
-      expect(screen.getByTestId('root')).toMatchInlineSnapshot(
```

Comment:
The goal was asserting that when no custom 404 page was provided the default was wrapped in the user root. Why are we removing this assertion?

Reply:
I used mock to test this behaviour, if `RouteMatchMock` is called than it means that the root layout is present

Reply:
I added `rootRouteComponentMock` and relevant checks to restore the previous checks